lib/idx/factory/baseApi.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+import { makeIdxState } from '../idxState';
+import { canProceed, proceed } from '../proceed';
+import { startTransaction } from '../startTransaction';
+import {
+  clearTransactionMeta,
+  createTransactionMeta,
+  getSavedTransactionMeta,
+  getTransactionMeta,
+  isTransactionMetaValid,
+  saveTransactionMeta
+} from '../transactionMeta';
+import { FlowIdentifier, BaseIdxAPI, OktaAuthIdxInterface, FlowSpecification } from '../types';
```

Comment:
See comment on /factory/api, but I am imaging this file as
```suggestion
import { FlowIdentifier, MinimalIDXAPI, MinimalOAuthInterfaceForIDX, FlowSpecification } from '../types';

export function createMinimalIdxAPI(sdk: MinimalOAuthInterfaceForIDX): MinimalIDXAPI {
```